### PR TITLE
Compilation with TypeScript

### PR DESCRIPTION
**Pull request by dzucconi**
This moves to compiling the packages with TypeScript into the esnext module format. We now also emit type definitions and source maps.

So ultimately this goes something like: 

Reads source SVG files from `src/svg` and compresses them with SVGO into `build/svg`, the build script then runs and emits custom tsx files for each one into `build/src`. Then we compile that with `tsc` into `dist` including the package.json for it. CI runs all this and `semantic-release` runs inside of the `dist` directory and publishes it as a package.

The git diff carnage here is mostly because I had compressed SVGs checked into the repo.

cc @noahschechter

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,36 +8,37 @@
   "main": "dist/index.js",
   "browserslist": "> 0.25%, not dead",
   "scripts": {
-    "build": "NODE_ENV=production yarn optimize && node scripts/build",
+    "clean": "rm -rf dist && rm -rf build",
+    "compile": "tsc -p . && cp build/src/package.json dist/package.json",
```
Comment (noahschechter) [inline package.json]: woo!

**Comment by dzucconi** [comment]
@noahschechter ^ try out 2.0.0?

**Comment by noahschechter** [comment]
Will do, thanks again!

**Pull request merged**